src/main/kotlin/platform/sponge/creator/Sponge8Template.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2021 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.sponge.creator
+
+import com.demonwav.mcdev.creator.buildsystem.BuildSystem
+import com.demonwav.mcdev.platform.BaseTemplate
+import com.demonwav.mcdev.util.MinecraftTemplates.Companion.SPONGE8_BUILD_GRADLE_TEMPLATE
+import com.demonwav.mcdev.util.MinecraftTemplates.Companion.SPONGE8_GRADLE_PROPERTIES_TEMPLATE
+import com.demonwav.mcdev.util.MinecraftTemplates.Companion.SPONGE8_MAIN_CLASS_TEMPLATE
+import com.demonwav.mcdev.util.MinecraftTemplates.Companion.SPONGE8_SETTINGS_GRADLE_TEMPLATE
+import com.demonwav.mcdev.util.MinecraftTemplates.Companion.SPONGE8_SUBMODULE_BUILD_GRADLE_TEMPLATE
+import com.intellij.openapi.project.Project
+
+object Sponge8Template : BaseTemplate() {
+
+    fun applyMainClass(
+        project: Project,
+        pluginId: String,
+        packageName: String,
+        className: String
+    ): String {
+        val props = mutableMapOf(
+            "PLUGIN_ID" to pluginId,
+            "PACKAGE" to packageName,
+            "CLASS_NAME" to className
+        )
+
+        return project.applyTemplate(SPONGE8_MAIN_CLASS_TEMPLATE, props)
+    }
+
+    fun applyBuildGradle(
+        project: Project,
+        buildSystem: BuildSystem,
+        config: SpongeProjectConfig
+    ): String {
+        val props = mutableMapOf(
+            "GROUP_ID" to buildSystem.groupId,
+            "PLUGIN_ID" to buildSystem.artifactId,
+            "PLUGIN_VERSION" to buildSystem.version,
+            // SpongeGradle 1.1.0 adds the -SNAPSHOT suffix itself
```

Comment:
this will probably:tm: change at some point in the future (just as a heads-up at the moment)